cadc-util/src/main/java/ca/nrc/cadc/auth/DelegationToken.java

```diff
@@ -190,40 +268,64 @@ public static DelegationToken parse(String text, String requestURI, ScopeValidat
     {
         String[] fields = text.split(FIELD_DELIM);
         String userid = null;
+        Set<Principal> principalSet = new HashSet<>();
         String proxyUser = null;
         Date expirytime = null;
         URI scope = null;
         String signature = null;
+        List<String> domains = new ArrayList<>();
         try
         {
             for (String field : fields)
             {
+
                 String key = field.substring(0, field.indexOf(VALUE_DELIM));
                 String value = field.substring(field.indexOf(VALUE_DELIM) + 1);
-                if (key.equalsIgnoreCase("userid"))
+                log.debug("key = value: " + key + "=" + value);
+
+                if (key.equalsIgnoreCase(principalLabels.get(HttpPrincipal.class.getSimpleName())))
```

Comment:
It could be true that all Principal types have a constructor that takes the name as a string... could this use reflection to find and call that ctor? Otherwise, support for new types has to be added in IdentityType and in here...

And other places unless... maybe a static factory method in IdentityType is feasible? 
``Principal IdentityType.create(String name, IdentityType type)``